components/tidb_query_expr/src/impl_time.rs

```diff
@@ -157,6 +157,11 @@ pub fn add_datetime_and_duration(
     datetime: &DateTime,
     duration: &Duration,
 ) -> Result<Option<DateTime>> {
+    if datetime.invalid_zero() {
+        return ctx
+            .handle_invalid_time_error(Error::incorrect_datetime_value(datetime))
+            .map(|_| Ok(None))?;
+    }
     let mut res = match datetime.checked_add(ctx, *duration) {
         Some(res) => res,
         None => return Ok(None),
```

Comment:
I think the check is supposed to be here.

Reply:
Sorry, didn't get your point. The check is moved here.

PTAL